src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageItemListViewModel.cs

```diff
@@ -640,21 +640,34 @@ private void TriggerStatusLoader()
 
         private async System.Threading.Tasks.Task ReloadPackageVersionsAsync()
         {
-            var result = await _backgroundLatestVersionLoader.Value;
-
-            await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-
-            LatestVersion = result;
-            Status = GetPackageStatus(LatestVersion, InstalledVersion, AutoReferenced);
+            try
+            {
+                NuGetVersion latestVersion = await _backgroundLatestVersionLoader.Value;
+                LatestVersion = latestVersion;
+                Status = GetPackageStatus(LatestVersion, InstalledVersion, AutoReferenced);
+            }
+            // Cancelled async operations inside of _backgroundLatestVersionLoader callpaths are expected to raise an InvalidOperationException.
+            // One shouldn't spam PostOnFailure since these are to be expected when UI interaction causes previous operations
+            // (searches, or metadata downloads or deprecationMetadata downloads, etc...) to no longer be needed.
+            catch (InvalidOperationException)
```

Comment:
What % does this exception type contribute to our spam?

Reply:
we had a side conversation, yes, i spun towards invalidoperationexception, while i should be cathing two others. will modify and send.

Reply:
It's kind of weird that InvalidOperationException is being thrown while this is supposed to be a cancellation

Is it thrown on our end?

In general, I think ignoring cancellation exceptions in every scenario might be a good idea.

Reply:
yes, that was my mistake. i've checked telemetry, and the two main ones are TaskCanceled and OperationCanceled. OperationCanceled is the base class, so the new commit just catches that.
i checked telemetry for all three callsites.